src/doc/src/reference/features.md

```diff
@@ -1,134 +1,464 @@
+version "2", it will disable the default features for all workspace members.
+
+The resolver is a global option that affects the entire workspace. The
+`resolver` version in dependencies is ignored, only the value in the top-level
+package will be used. If using a [virtual workspace], the version should be
+specified in the `[workspace]` table, for example:
+
+```toml
+[workspace]
+members = ["member1", "member2"]
+resolver = "2"
+```
+
+The default if the `resolver` is not specified is the value `"1"` which will
+use the original resolver behavior.
+
+[build-dependencies]: specifying-dependencies.md#build-dependencies
+[dev-dependencies]: specifying-dependencies.md#development-dependencies
+[resolver chapter]: resolver.md#feature-resolver-version-2
+[virtual workspace]: workspaces.md#virtual-manifest
+
+### Build scripts
+
+[Build scripts] can detect which features are enabled on the package by
+inspecting the `CARGO_FEATURE_<name>` environment variable, where `<name>` is
+the feature name converted to uppercase and `-` converted to `_`.
+
+[build scripts]: build-scripts.md
+
+### Required features
+
+The [`required-features` field] can be used to disable specific [Cargo
+targets] if a feature is not enabled. See the linked documentation for more
+details.
+
+[`required-features` field]: cargo-targets.md#the-required-features-field
+[Cargo targets]: cargo-targets.md
+
+### SemVer compatibility
+
+Enabling a feature should not introduce a SemVer-incompatible change. For
+example, the feature shouldn't change an existing API in a way that could
+break existing uses. More details about what changes are compatible can be
+found in the [SemVer Compatibility chapter](semver.md).
+
+Care should be taken when adding and removing feature definitions and optional
+dependencies, as these can sometimes be backwards-incompatible changes. More
+details can be found in the [Cargo section](semver.md#cargo) of the SemVer
+Compatibility chapter. In short, follow these rules:
+
+* The following is usually safe to do in a minor release:
+  * Add a [new feature][cargo-feature-add] or [optional dependency][cargo-dep-add].
+  * [Change the features used on a dependency][cargo-change-dep-feature].
+* The following should usually **not** be done in a minor release:
+  * [Remove a feature][cargo-feature-remove] or [optional dependency][cargo-remove-opt-dep].
+  * [Moving existing public code behind a feature][item-remove].
+  * [Remove a feature from a feature list][cargo-feature-remove-another].
+
+See the links for caveats and examples.
+
+[cargo-change-dep-feature]: semver.md#cargo-change-dep-feature
+[cargo-dep-add]: semver.md#cargo-dep-add
+[cargo-feature-add]: semver.md#cargo-feature-add
+[item-remove]: semver.md#item-remove
+[cargo-feature-remove]: semver.md#cargo-feature-remove
+[cargo-remove-opt-dep]: semver.md#cargo-remove-opt-dep
+[cargo-feature-remove-another]: semver.md#cargo-feature-remove-another
+
+### Feature documentation and discovery
```

Comment:
Most users are on the discovery side, but this section mainly talks about documenting features.

What if a feature is not documented? How do I find out about it? The discovery used to be a stumbling block for me. Now I just go straight for Cargo.toml and then search for cfg with the feature name in the code to see what it does. There may be a better way, though. Mentioning some of these shortcuts may help a novice user.